### PR TITLE
fix: remove unsafe pop of messages

### PR DESCRIPTION
**Pull request by lukealvoeiro**
Rewind to right before the last user message, and prompt them again

### CODE REVIEW

**Review by elenazherdeva — APPROVED** [review]

**Review by michaelneale — APPROVED** [review]
Looks good - I assume the "we have removed..." message still holds as exchange does it now?

File: src/goose/cli/session.py [hunk 1]
```diff
@@ -156,10 +156,10 @@ def run(self) -> None:
                 self.reply()  # Process the user message.
             except KeyboardInterrupt:
                 self.interrupt_reply()
-            except Exception:
+            except FailedToGenerateMessageError:
```
Comment (baxen) [inline src/goose/cli/session.py]: Left a related message in the other PR, I think a more generic exception catch does make sense here
https://github.com/square/exchange/pull/30#discussion_r1746395686

**Review by baxen — APPROVED** [review]
Looks great!

File: src/goose/cli/session.py [hunk 2]
```diff
@@ -156,10 +156,10 @@ def run(self) -> None:
...
+                # rewind to right before the last user message
+                self.exchange.rewind_to_last_user_message()
                 print(traceback.format_exc())
-                if self.exchange.messages:
-                    self.exchange.messages.pop()
                 print(
                     "\n[red]The error above was an exception we were not able to handle.\n\n[/]"
```
Comment (zakiali) [inline src/goose/cli/session.py:164]: Maybe update this print statement to say a bit more clear about clearing the state to the just before the previous user prompt?

**Review by zakiali — APPROVED** [review]

**Pull request merged**